src/Api/Parser/XmlParser.php

```diff
@@ -162,12 +165,10 @@ private function parse_timestamp(Shape $shape, $value)
 
     private function parse_xml_attribute(Shape $shape, Shape $memberShape, $value)
     {
-        $namespace = $shape['xmlNamespace']['uri']
-            ? $shape['xmlNamespace']['uri']
-            : '';
-        $prefix = $shape['xmlNamespace']['prefix']
-            ? $shape['xmlNamespace']['prefix']
-            : '';
+        // `xmlNamespace` can be modeled as a string.  When modeled as an object,
+        // it is guaranteed to have a `uri` field.
+        $namespace = $shape['xmlNamespace']['uri'] ?? $shape['xmlNamespace'] ?? '';
```

Comment:
Also a question mark like indicated in #3144.  This field value is modeled as an object with a required field of `uri` and an optional field of `prefix` in all public AWS models.  It's possible the test suites will be updated to match this.

Reply:
This change is no longer needed.